src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -126,13 +126,17 @@ public void run(ActionContext context) throws Exception {
 
     QueryJobConfiguration queryConfig = builder.build();
 
-    Job queryJob = bigQuery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
 
-    LOG.info("Executing SQL as job {}.", jobId.getJob());
-    LOG.debug("The BigQuery SQL is {}", config.getSql());
+    // Setting external retry strategy for BigQuery client due to BigQuery Client not retrying when a job clashes
+    // with another job, due to error being 400.
 
-    // Wait for the query to complete
-    queryJob.waitFor();
+    final String retryableStringPattern = "Retrying the job with back-off";
+    List<Function<BigQueryException, Boolean>> retryRules = new ArrayList<>();
+    retryRules.add(
+      (BigQueryException e) -> e.getCode() == 400
+        && (e.getMessage().contains(retryableStringPattern) || e.getReason().contains(retryableStringPattern))
```

Comment:
Are the methods on BigQueryException like getMessage() and getReason() guaranteed to be non null?